src/iceberg/schema_field.h

```diff
@@ -86,4 +86,16 @@ class ICEBERG_EXPORT SchemaField : public iceberg::util::Formattable {
   bool optional_;
 };
 
+template <typename T, typename... Args>
+inline SchemaField MakeRequiredField(int id, std::string name, Args&&... args) {
```

Comment:
Perhaps we can add an extra `doc` parameter but is not actually used before supporting `doc` to `SchemaField`. In this way, we can have full definition of those fields in this PR.

Reply:
Created https://github.com/apache/iceberg-cpp/pull/96